<launch>                                                                          

  <!-- Arguments of launch -->

  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" value="0.3"/>
  <arg name="yaw" default="0.0"/>
  <arg name="gui" default="false"/>
  <arg name="open_rviz" default="true" />
  <arg name="s3_bucket" default="$(optenv BUCKET_NAME your-s3-bucket-name)" />
  <arg name="model" default="$(optenv TURTLEBOT3_MODEL waffle_pi)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="gazebo_world" default="$(optenv GAZEBO_WORLD DEFAULT)" doc="model  type [DEFAULT, WORLDFORGE]" />

  <group if="$(eval gazebo_world=='DEFAULT')" >
    <include file="$(find aws_robomaker_small_house_world)/launch/small_house.launch">
      <arg name="gui" value="$(arg gui)"/>
    </include>
  </group>

  <group if="$(eval gazebo_world=='WORLDFORGE')" >
    <include file="$(find aws_robomaker_worldforge_worlds)/launch/launch_world.launch" />
  </group>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find simulation_app)/rviz/view_data.rviz" if="$(arg open_rviz)"/>

  <!-- Enable navigation system -->
  <include file="$(find simulation_app)/launch/turtlebot3_mapping_setup.launch">
    <arg name="x_pos" value="$(arg x_pos)"/>
    <arg name="y_pos" default="$(arg y_pos)"/>
    <arg name="z_pos" default="$(arg z_pos)"/>
    <arg name="yaw" default="$(arg yaw)"/>
    <arg name="model" default="$(arg model)"/>
  </include>

  <include file="$(find s3_file_uploader)/launch/s3_file_uploader.launch" >
    <arg name="s3_bucket" value="$(arg s3_bucket)"/>
  </include>

  <node pkg="simulation_app" type="file_uploader.py" name="file_uploader" >
       <param name="ROBOT_STOP_TIMEOUT"  type="double" value="60"/>
       <param name="TOTAL_MAPPING_TIMEOUT"  type="double" value="900"/>
       <param name="NORM_ONE_DISTANCE_THRESHOLD"  type="double" value="0.1"/>
       <param name="S3_PREFIX_PATH"  type="str" value="map_results/"/>
       <param name="LOCAL_WRITE_PATH"  type="str" value="/tmp/"/>
       <!-- Do not modify the simulation_job_id env variable. Its pulled automatically from RoboMaker Simulations -->
       <param name="AWS_ROBOMAKER_SIMULATION_JOB_ID"  type="str" value="$(optenv AWS_ROBOMAKER_SIMULATION_JOB_ID default_job_id)"/>
  </node>
</launch>
